### PR TITLE
ansible: update point releases and unify build

### PR DESCRIPTION
**Pull request by peterhoeg**
###### Motivation for this change

1) We had lots of copy paste - instead use a generic builder for the various versions.

2) Default version changed to latest (2.5 instead of 2.4) - ref #40015

3) Point release updates for all versions

Cc: @jgeerds @joamaki 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/admin/ansible/default.nix [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      homepage = http://www.ansible.com;
+      description = "A simple automation tool";
+      license = with licenses; [ gpl3 ] ;
+      maintainers = with maintainers; [ jgeerds joamaki ];
+      platforms = with platforms; linux ++ darwin;
+    };
+  };
+
+in rec {
+  # I really don't know why we carry so many versions...
```
Comment (Mic92) [inline pkgs/tools/admin/ansible/default.nix]: We should only have the official supported releases here, which is 2.2 up to 2.4 https://docs.ansible.com/ansible/2.4/release_and_maintenance.html#release-status

File: pkgs/tools/admin/ansible/default.nix [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    version = "2.4.4.0";
+    sha256  = "0n1k6h0h6av74nw8vq98fmh6q4pq6brpwmx45282vh3bkdmpa0ib";
+  };
+
+  ansible_2_5 = generic {
+    version = "2.5.2";
+    sha256  = "1r9sq30xz3jrvx6yqssj5wmkml1f75rx1amd7g89f3ryngrq6m59";
+  };
+
+  ansible2 = ansible_2_5;
```
Comment (Mic92) [inline pkgs/tools/admin/ansible/default.nix]: Should be 2.4. 2.5 is still in development.

Comment (peterhoeg) [reply pkgs/tools/admin/ansible/default.nix]: Are you 100% sure? 2.5 looks to be the stable release version on https://docs.ansible.com where 2.4 is listed as a "previous version".

Comment (Mic92) [reply pkgs/tools/admin/ansible/default.nix]: Ok. You are right.

**Comment by FRidh** [comment]
Could a second commit be added that removes the old versions?

File: pkgs/tools/admin/ansible/default.nix (hunk 1 above)
Comment (peterhoeg) [reply pkgs/tools/admin/ansible/default.nix]: Done

**Comment by peterhoeg** [comment]
> Could a second commit be added that removes the old versions?

Done

File: pkgs/tools/admin/ansible/default.nix [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      maintainers = with maintainers; [ jgeerds joamaki ];
+      platforms = with platforms; linux ++ darwin;
+    };
+  };
+
+in rec {
+  # We will carry all the supported versions
+
+  ansible_2_4 = generic {
+    version = "2.4.4.0";
```
[the remark this answers is not in the document]
Comment (peterhoeg) [reply pkgs/tools/admin/ansible/default.nix]: Version 2.1 built properly with python3 when I tried.

[the remark this answers is not in the document]
Comment (xeji) [reply pkgs/tools/admin/ansible/default.nix]: I'm not saying it doesn't work - it's just not officially supported. It was more of a question whether we want to package a "tech previer".

[the remark this answers is not in the document]
Comment (peterhoeg) [reply pkgs/tools/admin/ansible/default.nix]: Turns out that previously all of them were built with python2, so in order not to change too much at one time, I'll change it to that here as well.

**Comment by xeji** [comment]
@GrahamcOfBorg build ansible ansible_2_4

**Pull request merged**